service/storage_proxy.cc

```diff
@@ -158,6 +158,18 @@ class storage_proxy::remote {
     netw::connection_drop_slot_t _connection_dropped;
     netw::connection_drop_registration_t _condrop_registration;
 
+    std::optional<replica::stale_topology_exception> try_apply_fence(fencing_token token) const noexcept {
+        // topology_version == 0 means the caller is not aware about the fencing yet
+        const auto fence_version = _sp._shared_token_metadata.get_fence_version();
```

Comment:
Shouldn't this be done on the op's captured effective_replication_map_ptr?

Otherwise, by the time we capture it, it may change. Or if we captured it before, we may have captured an old one. Maybe this should be a method of effective_replication_map.

Reply:
The op should use only one token metadata version on a coordinator. But on a replica we need to make sure that by the time reply is sent the version the coordinator uses is still not fenced.

Reply:
We should be able to run with different-but-compatible topology versions.